### ISSUE TITLE
Fix the nav-bar while scrolling

### ISSUE THREAD

**Issue by pran-av** [Real-Dev-Squad/website-members#339]
### Aim: Ease of navigation for users

The header should remain fixed on the top of the page when a user scrolls. The constituents of the header remain the same.

Functionality:
• Shrink on scrolling
• Inflate to the original size on hovering at any point in the page

Please connect with me in case of any queries:
[Pranav Mandhare](mailto:pranavmandhare27@gmail.com)
Discord: Pranav#5274

**Comment by pallabez** [comment]
Hey, @pran-av can you link a demo video of how you want it to be?

**Comment by pran-av** [comment]
Hi @pallabez, sorry for the delay in getting back.

This article has a good video and instructions included for shrinking of the header: https://css-tricks.com/how-to-create-a-shrinking-header-on-scroll-without-javascript/

For sizing and opacity, we can take inspiration from https://basecamp.com/ header. Ex: Padding size of 2 rem seems good looking for the website. Matching the font sizes will also work.

The enhancements to do:
• Fix the header to the top of the website, so it sticks while scrolling
• While scrolling we remove the extra padding and make it sleeker for an enhanced experience (as in the first link). On hovering: we add the padding back.
Possible steps: Remove the individual text margins and padding. Replace by an external header container padding. Manipulate the header container padding.
• Reduce the Opacity of the header while scrolling. On hover: Return to full opacity.
Reference: Basecamp website's header. (Optional: execute if you think it is creating a substantial presentability)

Here's a video I recorded with some more details:
https://www.loom.com/share/e143b77191764640ac8221b7b0722fdb

Happy to connect on a Discord call if more clarification and discussion are required.

**Comment by pallabez** [comment]
Nicely explained. Thank you for the clear requirements. 

One problem I see here is we need to make the nav bar changes to each site that contains navbar. Check out the issue https://github.com/Real-Dev-Squad/website-status/issues/233

It can be done but might take upto 1-2 month to get all sites to do the same. I'd suggest to just fix the nav bar to the top without any resizing or opacity change, so that we can have a much gradual change.

**Comment by pran-av** [comment]
Yes. Went through the old PRs. Let's prioritize fixing the navbar to the screen right now.

We will perform future optimizations on separate issues.
